### ISSUE TITLE
Next.js 12.2.3 complains about `nx` in nextConfig

### ISSUE THREAD

**Issue by PatrikTheDev** [nrwl/nx#11297]
## Current Behavior

`nx` is getting passed into the `nextConfig` when using with-nx. Next.js 12.2.3 now complains about any key which isn't supposed to be included (a warning on every launch is thrown)

## Expected Behavior

`nx` shouldn't be passed to Next itself, it should be removed along the way.

### Environment
```
   Node : 16.16.0
   OS   : darwin x64
   yarn : 1.22.15
   
   nx : 14.4.3
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.4.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.4.3
   @nrwl/eslint-plugin-nx : 14.4.3
   @nrwl/express : Not Found
   @nrwl/jest : 14.4.3
   @nrwl/js : 14.4.3
   @nrwl/linter : 14.4.3
   @nrwl/nest : Not Found
   @nrwl/next : 14.4.3
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 14.2.0
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.4.3
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.4.3
   @nrwl/web : 14.4.3
   @nrwl/workspace : 14.4.3
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:
```

**Comment by indiejoseph** [comment]
same here, even upgraded to next(14.5.0-beta.5) version still doesn't work, the warning i got:
```
> nx run admin:serve:development

warn  - Invalid next.config.js options detected: 
[
  {
    "instancePath": "",
    "schemaPath": "#/additionalProperties",
    "keyword": "additionalProperties",
    "params": {
      "additionalProperty": "nx"
    },
    "message": "must NOT have additional properties"
  }
] 
```

**Comment by jaysoo** [comment]
Will be fixed in 14.5 https://github.com/nrwl/nx/pull/11313

**Comment by superhelten** [comment]
Is this still a problem? I can't figure out a way to add configs without getting something wrong at build.
